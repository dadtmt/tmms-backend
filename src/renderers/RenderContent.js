import R from 'ramda'
import React from 'react'
import { branch, renderComponent } from 'recompose'

import CharacterSheet from './CharacterSheet'
import Dice from './Dice'

const BaseComponent = () => <div />

const mayRenderDice = branch(
  R.propEq('type', 'dice'),
  renderComponent(Dice)
)

const mayRenderCharacterSheet = branch(
  R.propEq('type', 'characterSheet'),
  renderComponent(CharacterSheet)
)

export default R.pipe(mayRenderCharacterSheet, mayRenderDice)(BaseComponent)
