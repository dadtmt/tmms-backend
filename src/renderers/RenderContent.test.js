import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import RenderContent from './RenderContent'

it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(
    <RenderContent />,
    div
  )
})

it('renders dice if dice type', () => {
  const props = {
    details: 'details',
    made: false,
    master: false,
    modifier: 0,
    nbDices: 1,
    nbSides: 6,
    result: 6,
    type: 'dice'
  }
  expect(
    renderer.create(<RenderContent {...props} />).toJSON()
  ).toMatchSnapshot()
})

it('renders empty div on unknown type', () => {
  const props = {
    type: 'unknown'
  }
  expect(
    renderer.create(<RenderContent {...props} />).toJSON()
  ).toMatchSnapshot()
})
