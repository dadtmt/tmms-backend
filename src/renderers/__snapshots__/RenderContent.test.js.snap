exports[`test renders CharacterSheet if characterSheet type 1`] = `
<div
  className="panel panel-default"
  id={undefined}>
  <div
    className="panel-heading">
    Marom
  </div>
  <div
    className="panel-body">
    <span
      className="label label-default">
      Player sheet
    </span>
    <div
      className="well">
      Rodeur semi orc
    </div>
    <img
      className="img-rounded"
      src="https://s-media-cache-ak0.pinimg.com/736x/fe/a2/b2/fea2b2e27dc64aed4228bb3697018436.jpg" />
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-heading">
        Characteristics
      </div>
      <div
        className="panel-body">
        <table
          className="table">
          <tbody>
            <tr>
              <td>
                Force
                :
              </td>
              <td>
                17
              </td>
            </tr>
            <tr>
              <td>
                Dex
                :
              </td>
              <td>
                18
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      className="panel panel-default"
      id={undefined}>
      <div
        className="panel-heading">
        Gear
      </div>
      <div
        className="panel-body">
        <table
          className="table">
          <tbody>
            <tr>
              <td>
                Arc Feu
              </td>
              <td>
                1D8 + 1D6 feu
              </td>
              <td>
                <img
                  className="img-rounded"
                  src="https://s-media-cache-ak0.pinimg.com/236x/3b/76/10/3b76104f3387a68e7210c438de6a23c7.jpg"
                  width="50" />
              </td>
              <td>
                1
              </td>
            </tr>
            <tr>
              <td>
                Fleches
              </td>
              <td />
              <td />
              <td>
                10
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`test renders dice if dice type 1`] = `
<div>
  <span
    className="label label-default">
    Lancer les d√©s: 
    1D6+0
  </span>
</div>
`;

exports[`test renders empty div on unknown type 1`] = `<div />`;
